src/org.hdfgroup.hdfview/hdf/view/TableView/DataProviderFactory.java

```diff
@@ -1672,6 +1680,79 @@ public Object getDataValue(Object obj, int index)
 
             return theValue;
         }
+
+        /**
+         * update the data value of a compound type.
+         *
+         * @param columnIndex the column
+         * @param rowIndex    the row
+         * @param newValue    the new data value object
+         */
+        @Override
+        public void setDataValue(int columnIndex, int rowIndex, Object newValue)
+        {
+            Object theValue = newValue;
+            if (isFLT16)
+                theValue = Short.toString(Float.floatToFloat16(Float.parseFloat((String)newValue)));
```

Comment:
Why is this set as a string here rather than a real data value?

Reply:
Might be an artifact before the Display/Converter changes?

Reply:
Could be. I never created one, but it'd be useful to have a test file with a float16 member inside a compound, since this seems like editing one of those values would convert it to a String and either fail on write or cause weird issues later during read.

Reply:
I did test a change value and it worked - will need to check into container types.

Reply:
You can see the other `setDataValue` calls below return `Float.float16ToFloat((short)bufObject)` rather than a string. This almost certainly doesn't need the `Short.toString(` part and I'm surprised it doesn't cause problems.

Reply:
I'd assumed that this `setDataValue` call was for float16 inside a compound because of the Javadoc comment above, but it's actually the regular `setDataValue` call and that comment needs updating.

Reply:
Maybe -  I know not having it failed the first time I tried - will try again since changing other code.

Reply:
Needs to be a string - fails without it

Reply:
Then something is wrong elsewhere. For numerical values the DataProvider should be setting actual values and not strings. The only time numerical data should be working on a String is when converting to displayable formats in the DataDisplayConverters.

Reply:
The trouble is that float16 is a Short underneath and everything was triggering om that type.

Reply:
Right, I understand, but it shouldn't be a string. It's unfortunate that the implementation chose to encode the value in a `short`, because it forces us to implement "wrapper" `setDataValue` functions just to check if a value is a float16 value so that we can convert it from a `short` to a real float type, but I suppose it is what it is. But, the data array we work with should always be in terms of either `short` or `float` and then converted to `String` in the DataDisplayConverters and `float` elsewhere where we need a real value (if we didn't already choose to convert it to a `float` for the data array).